### PR TITLE
Feature3 enemy

### PR DESCRIPTION
**Pull request by Kundankmr500**

### CODE REVIEW

File: Assets/Scripts/Bullet/BulletController.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+        public void FireBullet(Transform bulletTransform, float bulletLaunchForce)
+        {
+            C_BulletView.bulletBody.velocity = bulletLaunchForce * bulletTransform.forward;
+        }
+
+        public void KillBullet()
+        {
+            Object.Destroy(C_BulletView.gameObject);
```
Comment (mayankgrover) [inline Assets/Scripts/Bullet/BulletController.cs:37]: destroy in view class

File: Assets/Scripts/Bullet/BulletView.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+{
+    [RequireComponent(typeof(Rigidbody))]
+    public class BulletView : MonoBehaviour
+    {
+        public LayerMask m_TankMask;                        
+        public ParticleSystem m_ExplosionParticles;         
+        public AudioSource m_ExplosionAudio;
+
+        public Rigidbody bulletBody;
+        private float m_MaxDamage = 10f;
```
Comment (mayankgrover) [inline Assets/Scripts/Bullet/BulletView.cs:18]: these should model

File: Assets/Scripts/Enemy/EnemyService.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using Generic;
+
+namespace Enemy
+{
+    public class EnemyService : MonoSingletonGeneric<EnemyService>
+    {
+        public Transform EnamyParent;
+        public List<EnemyController> Enemies = new List<EnemyController>();
```
Comment (mayankgrover) [inline Assets/Scripts/Enemy/EnemyService.cs:11]: private

File: Assets/Scripts/Enemy/EnemyService.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        public List<EnemyScriptableObj> EnamyScriptableObjs;
```
Comment (mayankgrover) [inline Assets/Scripts/Enemy/EnemyService.cs:12]: Resources.Find
Assets from the disk

File: Assets/Scripts/Bullet/BulletController.cs (hunk 1 above)
Comment (Kundankmr500) [reply Assets/Scripts/Bullet/BulletController.cs:37]: As you suggested to destroy in view component only, I have done this like below chain, but still, this is showing null reference sometimes, Am I creating the wrong chain?
            BulletView.KillView();
            BulletModel = null;
            BulletView = null;
            BulletParent = null;

File: Assets/Scripts/Enemy/EnemyService.cs (hunk 4 above)
Comment (Kundankmr500) [reply Assets/Scripts/Enemy/EnemyService.cs:12]: Sir, I think this is you explaining to someone query, right?

File: Assets/Scripts/Bullet/BulletView.cs (hunk 2 above)
Comment (Kundankmr500) [reply Assets/Scripts/Bullet/BulletView.cs:18]: Done!

File: Assets/Scripts/Enemy/EnemyService.cs (hunk 3 above)
Comment (Kundankmr500) [reply Assets/Scripts/Enemy/EnemyService.cs:11]: Done!

File: Assets/Scripts/Bullet/BulletController.cs (hunk 1 above)
Comment (Kundankmr500) [reply Assets/Scripts/Bullet/BulletController.cs:37]: Done!

File: Assets/Scripts/Enemy/EnemyService.cs (hunk 4 above)
Comment (Kundankmr500) [reply Assets/Scripts/Enemy/EnemyService.cs:12]: I think you mentioned this because someone asked the question here.

**Pull request merged**